app/finders/email_channel_finder.rb

```diff
@@ -7,7 +7,11 @@ def perform
     channel = nil
     recipient_mails = @email_object.to.to_a + @email_object.cc.to_a
     recipient_mails.each do |email|
-      channel = Channel::Email.find_by('lower(email) = ? OR lower(forward_to_email) = ?', email.downcase, email.downcase)
+      mail_id, domain = email.split('@')
+      original_mail_address = mail_id.split('+')[0]
+      channel = Channel::Email.where('lower(email) ~* ? OR lower(forward_to_email) ~* ?',
```

Comment:
Missed the case where we do .downcase on email. Pushing it alongside some refactoring.